src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1701,17 +1711,21 @@ bool
 LedgerMaster::shouldAcquire (
     std::uint32_t const currentLedger,
     std::uint32_t const ledgerHistory,
+    boost::optional<LedgerIndex> minSeq,
     std::uint32_t const ledgerHistoryIndex,
     std::uint32_t const candidateLedger) const
 {
 
     // Fetch ledger if it might be the current ledger,
-    // is requested by the advisory delete setting, or
-    // is within our configured history range
+    // is the greater of both the minimum persisted ledger, if any,
+    // and the minimum ledger for online deletion to retain, or
+    // is within our configured history range.
 
     bool ret (candidateLedger >= currentLedger ||
-        ((ledgerHistoryIndex > 0) &&
-            (candidateLedger > ledgerHistoryIndex)) ||
+        candidateLedger >= std::max(
```

Comment:
@mtrippled  and I discussed this, and this comment is used to capture our discussion. Here we are changing the default behavior of rippled. Before rippled would not go out there to grab gaps and now it will. if online delete is not enabled, users can see an increase in disk usage. I suggested a cfg enhancement to make this behavior opt-in, but I also think leaving gaps not desireable. I do think this has a doc impact @mDuo13 @nbougalis

Reply:
I believe that filling in the gaps is preferable in all cases. The disk impact is probably not noticeable on well-resourced servers and may be insignificant enough to get lost in the noise of ledger size variance in general. I would vote against making this a config option.

Reply:
Also, @mDuo13 I think it would simplify this document: https://xrpl.org/ledger-history.html
Particularly the part about "With Advisory Deletion." Going forward, that will be the normal case, not just with advisory deletion.